src/Microsoft.Extensions.Logging.Console/Internal/ConsoleLoggerProcessor.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Logging.Console.Internal
+{
+    public class ConsoleLoggerProcessor
+    {
+        private const int _maxQueuedMessages = 1024;
+        // Writing to console is not an atomic operation in the current implementation and since multiple logger 
+        // instances are created with a different name. Also since Console is global, using a static lock is fine. 
+        private static readonly object _lock = new object();
+
+        private IConsole _console;
+
+        private readonly BlockingCollection<LogMessageEntry> _messageQueue = new BlockingCollection<LogMessageEntry>(_maxQueuedMessages);
+        private readonly Task _outputTask;
+
+        public ConsoleLoggerProcessor()
+        {
+            RegisterForExit();
+
+            // Start Console message queue processor
+            _outputTask = Task.Factory.StartNew(
+                ProcessLogQueue,
+                this,
+                TaskCreationOptions.LongRunning);
+        }
+
+        public IConsole Console
+        {
+            get { return _console; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                _console = value;
+            }
+        }
+
+        public bool HasQueuedMessages => _messageQueue.Count > 0;
```

Comment:
Would make it internal till someone wants it public; else its exposing a impl detail that someone will take a dependency on...